### PR TITLE
Add query param to check if Blazor WASM app is under debug

### PR DESCRIPTION
**Pull request by captainsafia**
Uses a `_blazor_debug` query param flag to check if the application is currently under debug. For backwards-compat, this value defaults to true. For scenarios where it needs to be disabled, users will need to navigate to `https://localhost:5001?_blazor_debug=false` when they first visit the page.

We can update the `blazorwasm` VS Code debug experience to set this value correctly if start without debugging is set. We'll need to do the same in VS. I've pinged the VS debugging team about this.

Addresses https://github.com/dotnet/aspnetcore/issues/24623

### CODE REVIEW

**Comment by captainsafia** [comment]
@guardrex Heads up that this change will require documentation page updates.

File: src/Components/Web.JS/src/Platform/Mono/MonoDebugger.ts [hunk 1]
```diff
@@ -3,10 +3,17 @@ import { WebAssemblyResourceLoader } from '../WebAssemblyResourceLoader';
 const currentBrowserIsChrome = (window as any).chrome
   && navigator.userAgent.indexOf('Edge') < 0; // Edge pretends to be Chrome
 
+let isDebugging = true;
+
+window.addEventListener('load', () => {
+  const params = new URLSearchParams(window.location.search);
+  isDebugging = params.get('_blazor_debug') === 'true';
```
Comment (pranavkm) [inline src/Components/Web.JS/src/Platform/Mono/MonoDebugger.ts:10]: Should we call this `blazor_debug=true`? We would want users to possibly use this if they were debugging inside chrome no? Also, should we include this in the message we print as part of the Shift + Alt + D: https://github.com/dotnet/aspnetcore/blob/648c15dbe90fb8c113f7c6b4adeb40d9e10494f6/src/Components/WebAssembly/DebugProxy/src/TargetPickerUi.cs#L38?

Comment (captainsafia) [reply src/Components/Web.JS/src/Platform/Mono/MonoDebugger.ts:10]: > Should we call this blazor_debug=true? 

Do you mean remove the underscore at the start or set it to default to try?

Comment (captainsafia) [reply src/Components/Web.JS/src/Platform/Mono/MonoDebugger.ts:10]: I added the warning in the console when the user presses the keyboard shortcut. It's a little bit more straightforward to implement it there.

Comment (pranavkm) [reply src/Components/Web.JS/src/Platform/Mono/MonoDebugger.ts:10]: I was going to suggest doing both. We do want for users to be able to set this if they'd prefer using the browser. Using `_` feels like messing with private state.

File: src/Components/Web.JS/src/Platform/Mono/MonoDebugger.ts [hunk 2]
```diff
@@ -26,6 +33,8 @@ export function attachDebuggerHotkey(resourceLoader: WebAssemblyResourceLoader)
         console.error('Cannot start debugging, because the application was not compiled with debugging enabled.');
       } else if (!currentBrowserIsChrome) {
         console.error('Currently, only Microsoft Edge (80+), or Google Chrome, are supported for debugging.');
+      } else if (!isDebugging) {
+        console.error('_blazor_debug query parameter must be set to enable debugging.');
```
Comment (pranavkm) [inline src/Components/Web.JS/src/Platform/Mono/MonoDebugger.ts:37]: Could we print the url here? E.g. `To launch with debugging, use `${location.href}?blazor_debug=true`? I wouldn't put too much effort in making sure it does not have two query tokens

**Review by pranavkm — APPROVED** [review]

File: src/Components/Web.JS/src/Platform/Mono/MonoDebugger.ts (hunk 2 above)
Comment (captainsafia) [reply src/Components/Web.JS/src/Platform/Mono/MonoDebugger.ts]: Done!

**Pull request merged**